pkg/virt-handler/virt-chroot/virt-chroot.go

```diff
@@ -20,28 +20,58 @@
 package virt_chroot
 
 import (
+	"fmt"
+	"os"
 	"os/exec"
 	"strings"
 
+	ps "github.com/mitchellh/go-ps"
+
 	"kubevirt.io/kubevirt/pkg/safepath"
 	"kubevirt.io/kubevirt/pkg/unsafepath"
 )
 
 const (
 	binaryPath     = "/usr/bin/virt-chroot"
-	mountNamespace = "/proc/1/ns/mnt"
+	MountNamespace = "/proc/1/ns/mnt"
 )
 
+// getParentProcessByName - return the parent process
+// this is needed to handle the case kubernetes and kubevirt
+// running inside a container and the mount namespace is different
+// from the host/node.
+func getParentProcess() (ps.Process, error) {
+	pid := os.Getpid()
+	process, err := ps.FindProcess(pid)
+	if err != nil {
+		return nil, err
+	}
+
+	process, err = ps.FindProcess(process.PPid())
+	if err != nil {
+		return nil, err
+	}
+	return process, nil
+}
+
+func GetHostNamespacePath() string {
+	process, err := getParentProcess()
+	if err == nil {
+		return fmt.Sprintf("/proc/%d/ns/mnt", process.Pid())
+	}
+	return MountNamespace
```

Comment:
I think this can be simplified https://pkg.go.dev/os#Getppid:
```suggestion
	return fmt.Sprintf("/proc/%d/ns/mnt", os.Getppid())
```
Thus, you do not need to introduce `getParentProcess` and `MountNamespace = "/proc/1/ns/mnt"` can be dropped as well.

Reply:
@vasiliy-ul updated the patch. See if this works now with tests.